aws/resource_aws_ecs_capacity_provider_test.go

```diff
@@ -0,0 +1,320 @@
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcsCapacityProviderExists(resourceName, &provider),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1"),
+				),
+			},
+			{
+				ResourceName:      resourceName,
+				ImportStateId:     rName,
+				ImportState:       true,
+				ImportStateVerify: true,
+			},
+			{
+				Config: testAccAWSEcsCapacityProviderConfigTags2(rName, "key1", "value1updated", "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcsCapacityProviderExists(resourceName, &provider),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "2"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key1", "value1updated"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+			{
+				Config: testAccAWSEcsCapacityProviderConfigTags1(rName, "key2", "value2"),
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckAWSEcsCapacityProviderExists(resourceName, &provider),
+					resource.TestCheckResourceAttr(resourceName, "tags.%", "1"),
+					resource.TestCheckResourceAttr(resourceName, "tags.key2", "value2"),
+				),
+			},
+		},
+	})
+}
+
+// TODO add an update test config - Reference: https://github.com/aws/containers-roadmap/issues/633
+
+func testAccCheckAWSEcsCapacityProviderDestroy(s *terraform.State) error {
+	// Reference: https://github.com/aws/containers-roadmap/issues/632
+	return nil
+}
+
+func testAccCheckAWSEcsCapacityProviderExists(resourceName string, provider *ecs.CapacityProvider) resource.TestCheckFunc {
+	return func(s *terraform.State) error {
+		rs, ok := s.RootModule().Resources[resourceName]
+		if !ok {
+			return fmt.Errorf("Not found: %s", resourceName)
+		}
+
+		conn := testAccProvider.Meta().(*AWSClient).ecsconn
+
+		input := &ecs.DescribeCapacityProvidersInput{
+			CapacityProviders: []*string{aws.String(rs.Primary.ID)},
+			Include:           []*string{aws.String(ecs.CapacityProviderFieldTags)},
+		}
+
+		output, err := conn.DescribeCapacityProviders(input)
+
+		if err != nil {
+			return fmt.Errorf("error reading ECS Capacity Provider (%s): %s", rs.Primary.ID, err)
+		}
+
+		for _, cp := range output.CapacityProviders {
+			if aws.StringValue(cp.CapacityProviderArn) == rs.Primary.ID {
+				*provider = *cp
+				return nil
+			}
+		}
+
+		return fmt.Errorf("ECS Capacity Provider (%s) not found", rs.Primary.ID)
+	}
+}
+
+func testAccAWSECSCapacityProviderAutoScalingGroupConfig(rName string) string {
+	return fmt.Sprintf(`
+data "aws_ami" "test_ami" {
+	most_recent = true
+	owners      = ["amazon"]
+
+	filter {
+		name   = "name"
+		values = ["amzn-ami-hvm-*-x86_64-gp2"]
+	}
+}
+
+resource "aws_launch_configuration" "foobar" {
+	image_id      = "${data.aws_ami.test_ami.id}"
+	instance_type = "t2.micro"
+}
+
+resource "aws_autoscaling_group" "bar" {
+	availability_zones   = ["us-west-2a"]
```

Comment:
We should try to avoid hardcoding availability zones to allow the testing to be region and partition agnostic -- the `aws_availability_zones` data source can help. 👍 See also: https://github.com/terraform-providers/terraform-provider-aws/pull/11151#discussion_r357187750